react/components/SearchResultContainer.js

```diff
@@ -38,28 +36,46 @@ const MAP_SEPARATOR = ','
  * Search Result Component.
  */
 class SearchResultContainer extends Component {
-  getLinkProps = ({ opt, variables, isSelected, type, pageNumber }) => {
-    let { query, map, orderBy } = this.props.searchQuery.variables
-    if (variables) {
-      query = variables.query || query
-      map = variables.map || map
-      orderBy = variables.orderBy || orderBy
+  getLinkProps = ({ opt, type, isSelected, ordenation, pageNumber }) => {
+    const { path, rest, map, pagesPath } = this.props
+    let { variables: { orderBy } } = this.props.searchQuery
+    orderBy = ordenation || ordenation
+    return getPagesArgs(
+      opt.Name,
+      type,
+      opt.Link,
+      path,
+      rest,
+      { map, orderBy, pageNumber },
+      pagesPath,
+      isSelected
+    )
+  }
+
+  getCategories() {
+    const {
+      facetsQuery,
+      searchQuery: { variables: { query, map } },
+      pagesPath,
+    } = this.props
+    const { CategoriesTrees: tree } = facetsQuery.facets
+    const categories = getCategoriesFromQuery(query, map)
+    const category = findInTree(tree, categories, 0)
+    if (pagesPath === 'store/department') {
+      return tree[0].Children
+    } else if (category) {
+      return category.Children || tree[0].Children
     }
-    return getPagesArgs(opt, query, map, orderBy, isSelected, type, pageNumber)
+    return null
   }
 
   renderSearchFilters() {
     if (!this.props.facetsQuery || !this.props.facetsQuery.facets) return
 
-    const {
-      facetsQuery: { facets },
-      searchQuery,
-    } = this.props
-    const query = searchQuery.variables.query
-    const map = searchQuery.variables.map
-    const selecteds = this.getSelecteds(query, map)
-    const isDisabled =
-      this.countSelecteds(selecteds) === LIMIT_SELECTEDS_TO_DISABLE
+    let { facetsQuery: { facets } } = this.props
+    facets = { ...facets }
```

Comment:
What about use the spread inside the destructure or use a different name inside of it?